### PR TITLE
Added a breadcrumbs helper 

### PR DESCRIPTION
**Pull request by Misha7776**
**CHECK LIST**
- [x] specs added
- [x] specs passed
- [x] code coverage >90%
- [ ] DB indexes added if needed
- [ ] seed.rb file is updated if needed
- [x] PR is reviewed manually again (to make sure you have 100% ready code)
- [x] PR doesn't have conflicts with main branch
- [x] I've checked new feature as logged in and logged out user if needed
- [x] "let!" is used in specs
- [ ] screenshot of rails_best_practices and other tools added
- [ ] screenshot of feature is added if task has UI

### CODE REVIEW

File: app/views/account/shared/_breadcrumbs.html.erb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<ol class="breadcrumb">
+  <% nav.each do |n| %>
```
Comment (romaslmd) [inline app/views/account/shared/_breadcrumbs.html.erb]: home icon?

File: app/views/account/shared/_breadcrumbs.html.erb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+<ol class="breadcrumb">
```
Comment (romaslmd) [inline app/views/account/shared/_breadcrumbs.html.erb]: home icon?

File: app/helpers/breadcrumbs_helper.rb [hunk 3]
```diff
@@ -0,0 +1,9 @@
+module BreadcrumbsHelper
+  def render_breadcrumbs(breadcrumbs_hash)
+    @breadcrumbs ||= []
```
Comment (romaslmd) [inline app/helpers/breadcrumbs_helper.rb]: no need
`#each` method comes from Enumerable, so it's applicable for hashes too, which you'r using below

File: app/views/account/shared/_breadcrumbs.html.erb [hunk 4]
```diff
@@ -0,0 +1,9 @@
...
+  <% nav.each do |n| %>
+      <% if n.equal? nav.last %>
```
Comment (romaslmd) [inline app/views/account/shared/_breadcrumbs.html.erb]: you may pass `nil` for active page as you don't need a url
so this'd change to
```
li= link_to_if element[:path], element[:title], element[:path]
```

File: app/views/account/companies/show.html.slim [hunk 5]
```diff
@@ -4,13 +4,10 @@
...
-          ol.breadcrumb
-            li.breadcrumb-item
-              a href=root_path  User
-            li.breadcrumb-item
-              a href=account_companies_path  All Companies
-            li.breadcrumb-item.active
-              = @company.name
+          = render_breadcrumbs({ 'User' => root_path,
+                  'All companies' => account_companies_path,
+                  @company.name => nil,
```
Comment (romaslmd) [inline app/views/account/companies/show.html.slim]: could you follow some order and put this as last element

File: app/views/account/pages/index.html.slim [hunk 6]
```diff
@@ -2,6 +2,11 @@
   .row
     .col-md-12
       h1 All company pages
+  .row.breadcrumbs-top
+    .breadcrumb-wrapper.col-12
+      = render_breadcrumbs({ 'Home' => root_url,
+                @pages.first.company_name =>  account_company_path(@pages.first.company_id),
```
Comment (romaslmd) [inline app/views/account/pages/index.html.slim]: cmon.. `@pages.first`, and 2 times?

File: app/views/account/pages/index.html.slim [hunk 7]
```diff
@@ -2,6 +2,11 @@
   .row
     .col-md-12
       h1 All company pages
+  .row.breadcrumbs-top
+    .breadcrumb-wrapper.col-12
```
Comment (romaslmd) [inline app/views/account/pages/index.html.slim]: `.col-12` ?
also, always put bootstrap classes in the beginning of set of classes

File: app/views/account/pages/show.html.slim [hunk 8]
```diff
@@ -1,26 +1,31 @@
...
-  div
+.content-wrapper
+  .content-header.row
+    .content-header-left.col-md-6.col-12.mb-2
+      .content-header-title.mb-0
+        h2 = @page.title
+      .row.breadcrumbs-top
+        .breadcrumb-wrapper.col-12
+          = render_breadcrumbs({'Dashboard' => account_root_path,
+                                @page.company_name => account_company_path(@page.company_id),
```
Comment (romaslmd) [inline app/views/account/pages/show.html.slim]: `account_company_path(@page.company)` would be better

File: spec/helpers/breadcrumbs_helper_spec.rb [hunk 9]
```diff
@@ -0,0 +1,11 @@
+require 'spec_helper'
+
+describe BreadcrumbsHelper do
```
Comment (romaslmd) [inline spec/helpers/breadcrumbs_helper_spec.rb]: `type: :helper`

File: spec/helpers/breadcrumbs_helper_spec.rb [hunk 10]
```diff
@@ -0,0 +1,11 @@
...
+  describe '#page_breadcrumbs' do
+    it 'returns the right breadcrumb' do
+      expected_html = "<ul class=\"breadcrumb\"><li class=\"breadcrumb-item\">"\
+                    "<a href=\"http://test.host/account\">Dashboard</a></li></ul>"
```
Comment (romaslmd) [inline spec/helpers/breadcrumbs_helper_spec.rb]: change `"<a href=\"/account\">Dashboard</a></li></ul>` to `"<a href=\"#{account_root_path}\">Dashboard</a></li></ul>`

File: spec/helpers/breadcrumbs_helper_spec.rb [hunk 11]
```diff
@@ -0,0 +1,11 @@
...
+      expect(helper.render_breadcrumbs('Dashboard' => account_root_url)).to eq(expected_html)
```
Comment (romaslmd) [inline spec/helpers/breadcrumbs_helper_spec.rb]: use path instead of url

File: spec/helpers/breadcrumbs_helper_spec.rb [hunk 12]
```diff
@@ -0,0 +1,11 @@
+require 'spec_helper'
+
+describe BreadcrumbsHelper do
+  describe '#page_breadcrumbs' do
+    it 'returns the right breadcrumb' do
+      expected_html = "<ul class=\"breadcrumb\"><li class=\"breadcrumb-item\">"\
```
Comment (romaslmd) [inline spec/helpers/breadcrumbs_helper_spec.rb]: prefer `let` or `let!` for data mocks

File: spec/helpers/breadcrumbs_helper_spec.rb [hunk 13]
```diff
@@ -0,0 +1,11 @@
+require 'spec_helper'
```
Comment (romaslmd) [inline spec/helpers/breadcrumbs_helper_spec.rb]: `require 'rails_helper'`

File: app/controllers/account/pages_controller.rb [hunk 14]
```diff
@@ -26,4 +27,8 @@ def resource
   def collection
     current_user.companies.find(params[:company_id]).pages.by_rating
   end
+
+  def current_company
```
Comment (Kmalynss) [inline app/controllers/account/pages_controller.rb]: update https://github.com/Ruby-IF081/heroes_team_app/pull/109/files#diff-5fbc6024966caaf0a54efe28813715c9R27 to use ```current_company```

**Comment by Misha7776** [comment]
Issue #71

**Pull request merged**